feature/amazon/src/main/java/com/revenuecat/purchases/amazon/AmazonBilling.kt

```diff
@@ -359,6 +331,54 @@ internal class AmazonBilling constructor(
         }
     }
 
+    private fun queryAmazonPurchases(
+        filterOnlyActivePurchases: Boolean,
+        onSuccess: (Map<String, StoreTransaction>) -> Unit,
+        onError: (PurchasesError) -> Unit
+    ) {
+        executeRequestOnUIThread { connectionError ->
+            if (connectionError == null) {
+                purchaseUpdatesHandler.queryPurchases(
+                    onSuccess = onSuccess@{ receipts, userData ->
+                        val filteredReceipts = if (filterOnlyActivePurchases) {
+                            receipts.filter { it.cancelDate == null || it.cancelDate > dateProvider.now }
```

Comment:
we typically use `requestDate` as the reference date for determining whether something is active. 
Given that this is only for Offline Entitlements, I think using `now` is the best path, but maybe we should add an explicit comment so when we look back at this in 12 months we know _why_ we're doing `now` and not `requestDate` here?

Reply:
This will also be used on SDK configuration and app foreground to send unsynced purchases with the backend. But I believe that won't affect things here (we are not setting the expiration date for entitlements yet). This is the data from the Amazon store only.

There is a case that we should consider though. If:
- A user time travels in the device
- The user has cancelled the purchase but not expired yet

This could return incorrect information, and we might not sync the purchase with the backend... That's the biggest risk of this change, but seems like in that case, the purchase shouldn't be considered active. Wdyt @aboedo ?

Reply:
As for adding a comment, seems like calculating the expiration date is not the responsibility of this code, but I can add something so it's not confusing 👍

Reply:
> The user has cancelled the purchase but not expired yet
In this case, getting access is correct, right? it's what happens when you cancel auto renew in GP

> A user time travels in the device
yeah, that one is a bummer, but I don't think it's a super practical thing to do nowadays since everything uses SSL and TLS negotiation doesn't work while time traveling. wdty?

Reply:
> In this case, getting access is correct, right? it's what happens when you cancel auto renew in GP


Ah right, I meant both things happening simultaneously. This case on its own is pretty common and is already handled.

> yeah, that one is a bummer, but I don't think it's a super practical thing to do nowadays since everything uses SSL and TLS negotiation doesn't work while time traveling. wdyt?

True, it's not very common. I guess there is some light time traveling that can happen but not too long. So I think it's ok to keep as is 👍